app/controllers/rfid_controller.rb

```diff
@@ -26,25 +26,25 @@ def card_number
       end
     end
   end
-  
+
   def check_session(rfid)
     active_session = rfid.user.lab_sessions.where("sign_out_time > ?", Time.now)
+    new_location = PiReader.find_by(pi_mac_address: params[:mac_address])
```

Comment:
What should happen is no PiReader is found? I think you should return an error, but that breaks the old readers.... I think that's correct though

Reply:
At the top of `def card_number` I create a new PiReader if no PiReader with `params[:mac_address]` is found. This makes adding new readers to spaces very easy